src/helpers.ts

```diff
@@ -34,3 +44,34 @@ export function createResponseByCookie<T, U = unknown>(
         response,
     };
 }
+
+/**
+ * Create a mock where the responses are determined by the value of a specific cookie.
+ * @public
+ * @typeParam TMockCookieValue - Possible values for the cookie.
+ * @typeParam TResponse - The type of the response bodies.
+ * @param options - The options for the mock setup.
+ * @returns A complete mock.
+ */
+export function createMockByCookie<
```

Comment:
Could we add some testcases of this?

I'm thinking of direct unittests for the function where we call it with known arguments and verify it produces the correct mock (but also indirectly testing that the typings work as expected).